### PR TITLE
retester: adding metrics for counting retests

### PR DESCRIPTION
**Pull request by nesty156**
Release repo PR: https://github.com/openshift/release/pull/32797
[DPTP-3158](https://issues.redhat.com/browse/DPTP-3158)

/cc @hongkailiu

### CODE REVIEW

File: pkg/retester/retester.go [hunk 1]
```diff
@@ -45,6 +46,39 @@ type backoffCache struct {
...
+		prometheus.CounterOpts{
+			Name: "retests_for_base_sha",
+			Help: "Number of retests for base sha they have already happened.",
+		},
+		[]string{"pr"},
+	)
+	retestPRShaConter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "retests_for_pr_sha",
+			Help: "Number of retests for PR sha they have already happened.",
```
Comment (hongkailiu) [inline pkg/retester/retester.go:70]:
```suggestion
			Name: "retest_total",
			Help: "Number of retest command in total issued by the tool.",
```

File: pkg/retester/retester.go [hunk 2]
```diff
@@ -45,6 +46,39 @@ type backoffCache struct {
...
+		},
+		[]string{"pr"},
```
Comment (hongkailiu) [inline pkg/retester/retester.go:72]:
```suggestion
		[]string{"org", "repo"},
```

File: pkg/retester/retester.go [hunk 3]
```diff
@@ -45,6 +46,39 @@ type backoffCache struct {
 	logger         *logrus.Entry
 }
 
+var (
+	maxRetestsForShaAndBaseGauge = prometheus.NewGauge(
```
[the remark this answers is not in the document]
Comment (nesty156) [reply pkg/retester/retester.go:50]: Should I remove the Gauges?

[the remark this answers is not in the document]
Comment (hongkailiu) [reply pkg/retester/retester.go:50]: Yes, Please remove everyone except `retest_total`.

File: pkg/retester/retester.go [hunk 4]
```diff
@@ -344,6 +378,8 @@ func (b *backoffCache) check(pr tide.PullRequest, baseSha string, policy Reteste
 
 	record.RetestsForBaseSha++
 	record.RetestsForPrSha++
+	retestBaseShaConter.With(prometheus.Labels{"pr": record.BaseSha}).Inc()
+	retestTotalConter.With(prometheus.Labels{"org": string(pr.Author.Login), "repo": string(pr.Repository.NameWithOwner)}).Inc()
```
Comment (hongkailiu) [inline pkg/retester/retester.go:382]: This one should be `inc()` when the comment is added on the PR successfully.

File: pkg/retester/retester.go [hunk 5]
```diff
@@ -45,6 +46,21 @@ type backoffCache struct {
 	logger         *logrus.Entry
 }
 
+var (
+	retestTotalCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "retest_total",
+			Help: "Number of retest command in total issued by the tool.",
```
Comment (hongkailiu) [inline pkg/retester/retester.go:53]: command ---> commands

File: pkg/retester/retester.go [hunk 6]
```diff
@@ -379,6 +395,7 @@ func (c *RetestController) retestOrBackoff(pr tide.PullRequest) error {
 	case retestBackoffPause:
 		c.logger.Infof("%s: %s (%s)", prUrl(pr), "no comment", message)
 	case retestBackoffRetest:
+		retestTotalCounter.With(prometheus.Labels{"org": string(pr.Author.Login), "repo": string(pr.Repository.NameWithOwner)}).Inc()
```
Comment (hongkailiu) [inline pkg/retester/retester.go:398]: This is still too early. Please do it after successfully calling the GH api to create the PR.

File: pkg/retester/retester.go [hunk 7]
```diff
@@ -45,6 +46,21 @@ type backoffCache struct {
 	logger         *logrus.Entry
 }
 
+var (
+	retestTotalCounter = prometheus.NewCounterVec(
```
Comment (hongkailiu) [inline pkg/retester/retester.go:50]: retestTotalCounter ---> retestTotal

File: pkg/retester/retester.go [hunk 8]
```diff
@@ -352,6 +368,8 @@ func (c *RetestController) createComment(pr tide.PullRequest, cmd, message strin
 	comment := fmt.Sprintf("%s\n\n%s\n", cmd, message)
 	if err := c.ghClient.CreateComment(string(pr.Repository.Owner.Login), string(pr.Repository.Name), int(pr.Number), comment); err != nil {
 		c.logger.WithField("comment", comment).WithError(err).Error("failed to create a comment")
+	} else if cmd == "/retest-required" {
+		retestTotal.With(prometheus.Labels{"org": string(pr.Author.Login), "repo": string(pr.Repository.NameWithOwner)}).Inc()
```
Comment (hongkailiu) [inline pkg/retester/retester.go:372]: pr.Author.Login ---> pr.Repository.Owner.Login
pr.Repository.NameWithOwner ---> pr.Repository.Name

**Review by hongkailiu — APPROVED** [review]

**Pull request merged**